### PR TITLE
Use the new shiny tile-decorator

### LINKED ISSUE #2: Join lines that share specified properties
Features with the same properties within a tile should be joined to avoid the overhead of having distinct geometries. Ideally this is exposed as an option that is passed to the decorator.

For example: `{ linejoin: true, linejoinUniqueBy: [ 'speedClass', 'FRC' ] }`

@mourner - Can you grab this?

cc @yhahn

### LINKED ISSUE #3: All numbers are encoded as floats
All number values are [encoded as floats](https://github.com/mapbox/tilelive-decorator/blob/master/index.js#L195). This has two consequences:
- integers are not efficiently packed using varint, leading to bulkier size
- bigger integers loose precision when packed as 32-bit floats

Instead integers should be encoded as either `uint` or `sint` depending on sign.

cc @aaronlidman

### PR DESCRIPTION
**Pull request by mourner**
Switches to using the new [tile-decorator module](https://github.com/mapbox/tile-decorator).

Requires additional `keepKeys` param in the url (e.g. `keepKeys=foo,bar`). Features:
- all keys/values that aren't in `keepKeys` are removed
- features with the same type + tags are merged into one feature with multiple parts, closes #2
- parts within one feature are sorted by proximity for better VT compression
- fixes a bug with number encoding, closes #3

cc @aaronlidman @willwhite @mick

### CODE REVIEW

**Comment by willwhite** [comment]
Nice! :bow: 

> all keys/values that aren't in keepKeys are removed

What's the use case for this versus removing the keys in the source (undecorated) tiles? Is this for if you want to remove the `key` field that's used for joining?

**Comment by mourner** [comment]
> Is this for if you want to remove the `key` field that's used for joining?

Yes, in addition to the case where you don't have easy control over undecorated tiles. I could make it optional but it'll require some more code in `tile-decorator`, so I suggest keeping it required for now if no one objects.

**Comment by willwhite** [comment]
:+1: no objections from me!

File: package.json [hunk 1]
```diff
@@ -7,22 +7,36 @@
...
+    "nyc": "^5.5.0",
     "tape": "^4.2.0",
-    "tilelive-s3": "^3.0.0"
+    "tilelive-s3": "^6.1.0",
+    "vector-tile": "^1.1.3"
   },
   "dependencies": {
     "hiredis": "^0.4.1",
-    "lru-cache": "^2.7.0",
-    "pbf": "^1.3.4",
```
Comment (aaronlidman) [inline package.json:23]: pbf is still used in test/index.js

Comment (mourner) [inline package.json:23]: right, readded

**Comment by mick** [comment]
awesome! :clap:

**Comment by aaronlidman** [comment]
Awesome @mourner this is fantastic.

**Pull request merged**